### PR TITLE
Update to react basic 6

### PR DESCRIPTION
**Pull request by joelmccracken**

### CODE REVIEW

**Comment by joelmccracken** [comment]
Hi! I think this is ready for review. As far as I can tell, it works correctly.

I'm not sure what is going on with the PR checks, though.

File: package.json
Comment (f-f) [inline package.json]: Is the downgrading of React accidental or was it to make something work? If not, let's upgrade it to the newest version

File: packages.dhall
Comment (f-f) [inline packages.dhall]: I'd prefer to point it to some more specific branch, or even better to a commit hash

File: src/Todo/App.purs
Comment (f-f) [inline src/Todo/App.purs]: What does `Kbd` stand for?

File: src/Todo/App.purs
Comment (f-f) [inline src/Todo/App.purs:105]: This is the part that saves to `localStorage` after every update right? It's quite neat :)
Let's move here the comment about saving to `localStorage` after every modification of the state, so it's clearer that's happening here

File: src/Todo/App.purs
Comment (f-f) [inline src/Todo/App.purs]: Style suggestion: I'd write this lambda as

```purescript
\case ->
  Just "Enter" -> KbdNewTodo
  _            -> Noop
```

Comment (f-f) [reply src/Todo/App.purs]: Logic-wise: why not keep the guard for `String.null` here? In this way we have to check for `Noop` in two cases (here, and inside the `KbdNewTodo` action, where we send a `NoUpdate` in case the guard is not satisfied)

File: src/Todo/App.purs
Comment (f-f) [inline src/Todo/App.purs]: Let's try to keep formatting consistent: I like how you align `if`, `then`, `else` in the next branch below, so we could do the same here

File: src/Todo/Task.purs
Comment (f-f) [inline src/Todo/Task.purs]:
```suggestion
```

File: src/Todo/Task.purs
Comment (f-f) [inline src/Todo/Task.purs]:
```suggestion
          KeyDown key ->
```

File: src/Todo/Task.purs
Comment (f-f) [inline src/Todo/Task.purs]:
```suggestion
```

File: src/Todo/App.purs
Comment (f-f) [inline src/Todo/App.purs]:
```suggestion
import React.Basic.DOM.Events ( key, targetChecked, targetValue)
```

File: src/Todo/Footer.purs
Comment (f-f) [inline src/Todo/Footer.purs]: Do we have a use for both `comp` and `component` or we can absorb `comp` in here?

File: index.js
Comment (f-f) [inline index.js]: I'd like to leave this file in js, so there's a clear point of insertion if you need to mix vanilla React components in js (as this is supposed to be some kind of starter app)

**Review by f-f — COMMENTED** [review]
Looks good! I left a bunch of minor comments, but they should all be fairly quick stuff. Other than that, it's basically ready to ship 👏 

Re build failing: you can see the results of the build by clicking on `Details` for the `deploy-preview` task (the last one). It says that it cannot find some module. Now, I can build this locally, so I suspect it's some caching issue, so I'll try clearing the cache and retrying

File: src/Todo/App.purs
Comment (joelmccracken) [reply src/Todo/App.purs]: It stands for Keyboard, meaning keyboard input. Maybe it doesn't make the most sense, I was a bit unsure what to name it.

File: src/Todo/App.purs
Comment (joelmccracken) [reply src/Todo/App.purs:105]: Exactly, this is what updates it =) It was easy to implement too. 

I'll move the comment.

File: src/Todo/Footer.purs
Comment (joelmccracken) [reply src/Todo/Footer.purs]: IIUC the note here: 

https://pursuit.purescript.org/packages/purescript-react-basic/6.2.0/docs/React.Basic#v:createComponent

then createComponent should always be created/specified at the module level. I had thought about doing `makeStateless (createComponent "Footer") render`, but that note made me hesitate 

Should I rename these, though? Do you think I understood the docs correctly?

File: index.js
Comment (joelmccracken) [reply index.js]: Gotcha, makes sense. I'll look at doing that. I went this route because this is the way I saw other examples doing it (well, not this *exactly*, but I mean having more of the bootstrapping done in PS. But at the time I didn't understand the APIs very well, so maybe now it will be easier to stitch everything together from JS.

File: packages.dhall
Comment (joelmccracken) [reply packages.dhall]: I set this to a specific branch, however I was not able to get it to work by setting version to a specific hash. Output in this case says for `spago install`:

```
...
Failed to install dependency "react-basic"

Git output:
error: Server does not allow request for unadvertised object 8ef0ba7879944c3fff98260f3844df4d8c7ec363

spago: Installation failed.
```

Comment (f-f) [reply packages.dhall]: @justinwoo I think the hash ref works only sometimes in spago right? (Or was it never supposed to work?)

File: src/Todo/Footer.purs
Comment (f-f) [reply src/Todo/Footer.purs]: Oh right, I missed these docs, thanks 👍 

Let's rename them for clarity:
- `comp` → `component`
- `component` → `footer`

File: src/Todo/App.purs
Comment (f-f) [reply src/Todo/App.purs]: `CreateNewTodo`? Or `Submit`, or `Add`

File: packages.dhall
Comment (joelmccracken) [reply packages.dhall]: It seems like it would be a good feature!

File: src/Todo/App.purs
Comment (joelmccracken) [reply src/Todo/App.purs]: Submit sounds good to me, will change

File: package.json
Comment (joelmccracken) [reply package.json]: Upgraded!

File: packages.dhall
Comment (joelmccracken) [reply packages.dhall]: Going to resolve for now, but if we figure out how to use a git ref I will change!

File: src/Todo/App.purs
Comment (joelmccracken) [reply src/Todo/App.purs]: The only reason why I did that is because it seemed to me that the concepts were sightly different: one check was for the user intending to submit the todo, whereas the other guard was validation that the todo can be created. But you're right, this does simplify, I'll change.

Comment (joelmccracken) [reply src/Todo/App.purs]: It seems that guard statements cannot be used with LambdaCase. I decided to stay with the more verbose case statement and use the guard like the original version.

**Comment by joelmccracken** [comment]
@f-f I believe I have handled everything you requested =)

**Comment by f-f** [comment]
@joelmccracken fantastic! 🎉

You should now have write access, so you can merge yourself :)

**Review by f-f — APPROVED** [review]
@joelmccracken fantastic! 🎉

You should now have write access, so you can merge yourself :)

**Comment by joelmccracken** [comment]
TY! I'm not sure if i'll get to this today, but I'm going to rebase the commits so that they're a little cleaner before merging.

**Comment by f-f** [comment]
Thank you! :)

**Pull request merged**
